snappi_ixnetwork/trafficitem.py

```diff
@@ -182,18 +284,311 @@ class TrafficItem(CustomField):
+            # with no url found, when the first response is 202 (in-progress)
+            # its keep checking the status of the url with 1 sec sleep, and
+            # after a while error is thrown. but could see the configuration
+            # applied at Ixnetwork. (Need to check with Eng team)
+            response = self._api._request("POST", url=url, payload=payload)
+        except Exception:
+            return
+        if (
+            response["result"].get("errata") is not None
+            and response["result"]["errata"] != []
+        ):
+            self._api.get_json_import_errors()
+            raise SnappiIxnException(
+                400, "{}".format(response["result"]["errata"])
+            )
+
+    def get_ports_encap(self, config):
+        ixn = self._api.assistant._ixnetwork
+        myfilter = [{"property": "name", "regex": ".*"}]
+        url, payload = self._get_search_payload(
+            "/",
+            "(?i)^(vport)$",
+            ["name"],
+            myfilter,
+        )
+        result = ixn._connection._execute(url, payload)
+        vports = {}
+        for vp in result:
+            if vp.get("vport") is None:
+                continue
+            for v in vp["vport"]:
+                vports[v["name"]] = v["xpath"]
+        return vports
+
+    def get_device_encap(self, config):
+        if len(config.devices) == 0:
+            return {}
+        dev_names = []
+        for f in config.flows:
+            if f.tx_rx.choice == "port":
+                continue
+            dev_names.extend(f.tx_rx.device.tx_names)
+            dev_names.extend(f.tx_rx.device.rx_names)
+        dev_names = list(set(dev_names))
+        if dev_names == []:
+            return {}
+        selects = []
+        url = None
+        for name in dev_names:
+            href = self._api.get_ixn_href(name)
+            if href is None:
+                # TODO change to more detailed message
+                raise Exception(
+                    "href is not found for device name {}".format(name)
+                )
+            href = href.split("ixnetwork")[-1]
+            query = "^(xpath|ethernet|ipv4|ipv6)$"
+            if "ipv4" in href or "ipv6" in href:
+                query = "^(xpath)$"
+            elif "networkGroup" in href:
+                query = "^(xpath|networkGroup|ipv4PrefixPools|ipv6PrefixPools)"
+            url, search = self._get_search_payload(href, query, ["name"], [])
+            selects.extend(search["selects"])
+        payload = {"selects": selects}
+        ixn = self._api.assistant._ixnetwork
+        result = ixn._connection._execute(url, payload)
+        if len(dev_names) != len(result):
+            # TODO change to more detailed message
+            raise Exception("search query result miss match with device names")
+        paths = {}
+        for i, d in enumerate(dev_names):
+            paths[d] = {"xpath": result[i]["xpath"]}
```

Comment:
Personally I preferred use self._api.get_device_encap. Because, we are planning to incorporate ISIS as well as unconnected interface. But if you filled searching is ok then I am fine.